### ISSUE TITLE
Multientry index does not work (sembast backing)

### ISSUE THREAD

**Issue by peter-fingo** [tekartik/idb_shim.dart#4]
Repro:

```
import 'dart:io';

import 'package:test/test.dart';
import 'package:idb_shim/idb.dart';
import 'package:idb_shim/idb_io.dart';

void main() {
  test('Simple query.', () async {
    final tmpDir = Directory('/tmp').createTempSync();
    IdbFactory idbFactory = getIdbSembastIoFactory('${tmpDir.path}/db');
    // open the database
    Database db = await idbFactory.open("my_records.db", version: 1,
        onUpgradeNeeded: (VersionChangeEvent event) {
      Database db = event.database;
      // create the store
      final store = db.createObjectStore('test', autoIncrement: true);
      store.createIndex('index', 'spath', multiEntry: true);
    });

    // put some data
    final object = <String, dynamic>{
      "spath": [1, 2]
    };
    var txn = db.transaction('test', "readwrite");
    var store = txn.objectStore('test');
    await store.put(object);
    await txn.completed;

    // read some data
    txn = db.transaction('test', "readonly");
    store = txn.objectStore('test');
    final cursorStream = store
        .index('index')
        .openCursor(range: KeyRange.lowerBound(0), autoAdvance: true)
        .map((cv) => cv.value);
    expect(cursorStream, emits(object));
  });
}

```

**Comment by alextekartik** [comment]
Thanks for the report. I'll take a look

**Comment by alextekartik** [comment]
I confirm that indeed for now multiEntry is not supported when using sembast backing. I'll tackle that hopefully by the end of the week.
Thanks for the detailed report

**Comment by alextekartik** [comment]
I implemented multiEntry support for sembast implementation. Published in 1.7.4-dev.0. Let me know if that works for you and thank you for the unit test

**Comment by peter-fingo** [comment]
Looks good, thanks!
